core/crypto/crypto_store/crypto_store_impl.cpp

```diff
@@ -96,40 +96,29 @@ namespace kagome::crypto {
   }
 
   CryptoStoreImpl::CryptoStoreImpl(
+      Path keys_directory,
       std::shared_ptr<ED25519Provider> ed25519_provider,
       std::shared_ptr<SR25519Provider> sr25519_provider,
       std::shared_ptr<Secp256k1Provider> secp256k1_provider,
       std::shared_ptr<Bip39Provider> bip39_provider,
       std::shared_ptr<CSPRNG> random_generator)
-      : keys_directory_{},
+      : keys_directory_(std::move(keys_directory)),
         ed25519_provider_(std::move(ed25519_provider)),
         sr25519_provider_(std::move(sr25519_provider)),
         secp256k1_provider_(std::move(secp256k1_provider)),
         bip39_provider_(std::move(bip39_provider)),
         random_generator_(std::move(random_generator)) {
+    BOOST_ASSERT_MSG(boost::filesystem::exists(keys_directory_),
+                     "keys directory doesn't exist");
+    BOOST_ASSERT_MSG(boost::filesystem::is_directory(keys_directory_),
+                     "keys directory path is not a valid directory");
```

Comment:
It will not doing on Release build.
If you want to check it, do it over AppStateManager ('prepare' stage).

Reply:
Other way for that is checking it within lambda in injector.

Reply:
it was decided to do without AppStateManager.